tests/integration_test.rs

```diff
@@ -144,6 +146,9 @@ async fn test(config_path: &'static str, t: Type) -> Result<()> {
     server_shutdown_tx.send(true)?;
     let _ = tokio::join!(server);
 
+    // Wait for the server connection to be closed (quic)
+    time::sleep(Duration::from_millis(2500)).await;
```

Comment:
Because quinn actually wraps the udp socket in the arc and puts it in the tokio task to run. So after the endpoint drop, the udp socket will not drop immediately. If you start the server again at this time, you will get the port reuse error.

Reply:
Then this is an issue for the hot-reloading feature. If a full restart is trigger, the new instance will try to bind on the same address and fail, which causes rathole to abort.

Any way to shutdown the UDP socket immediately, along with the server?

Reply:
I didn't find the relevant method in the source code and documentation, but I can send an issue to ask.

Reply:
Some bug fixes has been merged into the main branch. I don't know if that's relevant but maybe you want to do a rebase.

Reply:
ok, I am taking an exam these few days, I will continue to do this later.

Reply:
I suspect there's a short interval for all tokio tasks to shutdown, but normally that's not an issue because every CI that has been run so far is successful. 2.5s is too long for that. Anyway, I will find time to investigate this in the original code.